toplevel/coqloop.ml

```diff
@@ -330,17 +330,33 @@ let cproof p1 p2 =
 
 let drop_last_doc = ref None
 
+(* todo: could add other Set/Unset commands, such as "Printing Universes" *)
+let print_anyway_opts = [
+  [ "Diffs" ];
+  ]
+
+let print_anyway c =
+  let open Vernacexpr in
+  match c with
+  | VernacExpr (_, VernacSetOption (_, opt, _))
+  | VernacExpr (_, VernacUnsetOption (_, opt)) ->
+    List.mem opt print_anyway_opts
+  | _ -> false
+
 (* We try to behave better when goal printing raises an exception
    [usually Ctrl-C]
 
    This is mostly a hack as we should protect printing in a more
    generic way, but that'll do for now *)
-let top_goal_print oldp newp =
+let top_goal_print ~doc c oldp newp =
   try
     let proof_changed = not (Option.equal cproof oldp newp) in
-    let print_goals = not !Flags.quiet &&
-                      proof_changed && Proof_global.there_are_pending_proofs () in
-    if print_goals then Printer.print_and_diff oldp newp;
+    let print_goals = proof_changed && Proof_global.there_are_pending_proofs () ||
+                      print_anyway c in
+    if not !Flags.quiet && print_goals then begin
```

Comment:
Why has the `quiet` flag been moved out of the main condition?

Reply:
It seemed clearer to me this way.  NP to change.

Reply:
I'm OK with the current version.